src/managers/RoleManager.js

```diff
@@ -123,6 +123,18 @@ class RoleManager extends BaseManager {
       });
   }
 
+  /**
+   * Gets the managed role a user created when joining the guild, if any
+   * <info>Only ever available for bots</info>
+   * @param {UserResolvable} user The user to access the bot role for
+   * @returns {?Role}
+   */
+  botRoleFor(user) {
+    user = this.client.users.resolve(user);
+    if (!user || !user.bot) return null;
```

Comment:
might this return null 'falsely' for a partial user?

Reply:
removed bot check in https://github.com/discordjs/discord.js/pull/4628/commits/066eaa0d6104eb00ee880ed8f42c07400e5069a0